Community Guidelines Update (#97)
nim-works/nimskull: CODE_OF_CONDUCT.md

which promote diverse sets of skills and perspectives [^1] [^2] [^3] [^4]. Meritocracy is not
sustainable outside of interactions between absolute equals, including cultural
norms and understandings [^2].

</details>

<p align="right">(<a href="#top">back to top</a>)</p>

## Scope

This Code of Conduct applies to spaces both online and offline.

<details>

Online spaces managed by the Nim-Works organization will be held to the
following standards. Spaces that represent our users, or where users are
otherwise representing Nim-Works, may still be held accountable to these
standards depending upon context and official status of the representative.

Offline spaces where members interact, including but not limited to
conferences, will be similarly held to the standards below.

</details>

<p align="right">(<a href="#top">back to top</a>)</p>

## Code of Conduct

Outlined below are the main values of the CoC, each of which can be expanded to
show contributing rules and themes.

1. <details><summary>We are an open and inclusive community, so I should behave in a manner which is respectful of others without discrimination.</summary><br />

>    1. Discrimination against an individual(s) or group(s) damages the community and attacks the goals of our contributions; we specifically will not discriminate based on:
>       1. Virtue of race or ethnicity; country of origin, or country of residence; gender(s), biological sex or sexual orientation; age or experience; economic standing or education; disability - physical or otherwise. Not listed are not excluded.

Reviewer: ```suggestion
   1. Discrimination against an individual(s) or group(s) damages the community and attacks the goals of our contributions; we specifically will not discriminate based on, but not limited to:
      1. Virtue of race or ethnicity; country of origin, or country of residence; gender(s), biological sex or sexual orientation; age or experience; economic standing or education; disability - physical or otherwise.
```

Flow better this way IMO.

   2. Harassment - online or otherwise are not tolerated. This includes any offensive, belittling or threatening behaviour toward an individual(s) or group(s).
      1. Cultural diversity and language barriers may play a critical role in differentiating whether a perpetrator intentionally meant to be aggressive in their behaviour. Patience is crucial in investigating these cases.

</details>

> 2. <details><summary>Contribution does not entitle me or anyone else to act in any way that is not conducive to making this a productive and enjoyable environment</summary><br />

Reviewer: ```suggestion
2. <details><summary>Contribution does not entitle anyone to act in any way that is not conducive to making this a productive and enjoyable environment</summary><br />
```

Mostly a nit, I don't like how "me" is used here, it doesn't feel right, especially when we were using "we" earlier.

Reviewer 2: I almost suggested the same change, but the meaning is subtlety different or at least slightly more explicit of written as is.

The current style ensures that it's taken to mean as no "shield" (defense against restriction of action) is granted by way of contribution.

While if someone that has contributed could be entitled to a "sword" (offense ability to restrict others behavior).

The suggested change reads as too strong a shield.

Author: While I can't remember the intention behind writing this as it is, I do agree with Saem's interpretation. I prefer it this way or rephrased as:

"Contribution does not entitle we, nor anyone else, to act in any way that is not conducive to making this a productive and enjoyable environment"

    1. The cost toward the community for onboarding new members should not be expected to be unreasonable. We will provide the resources and guidance for members to learn, however we cannot be expected to provide a teaching service.
       1. Everyone starts at a different point and we will be patient in providing resources
       2. Simple answers such as "look at the manual" are not tolerable; these questions and their answers will likely be viewed again in the future by users through search engines and should therefore be treated appropriately.
       3. However, we recognise that it is not reasonable for repeated contributions to be fixed without any effort from the contributor to remedy their mistakes.
    2. First impressions matter; we are a good community who will welcome and share with those who are willing to learn.
       1. Failure to adapt can lead to an unreasonable cost to the community to maintain membership.

>        2. This may create situations which may seem contradictory. I understand however that this is in recognition of Code of Conduct item 4.

Reviewer: I think we should not use "I" here, given that the document speaks for the community.

</details>

4. <details><summary>We understand the expectations of us to act according to values and not technicalities.</summary><br />

>     1. The intolerable will never be tolerable. We will play by the rules as intended. Where it is not seen as discriminatory, people who try to excuse poor behaviour as not being written clearly within the Code of Conduct will be treated as if it was

Reviewer: ```suggestion
    1. The intolerable will never be tolerated. We will play by the rules as intended. Where it is not seen as discriminatory, people who try to excuse poor behaviour as not being written clearly within the Code of Conduct will be treated as if it was
```

And, uh, this sentence is not completed?

Reviewer 2: The tenses match in the original, so it's ok. The sentence is complete, just a sightly unusual ending. Needs a full-stop.

    2. Silence can sometimes be worse than agreeing with a harassor/perpetrator.
    3. If you are concerned due to some connection to the aggressor and do not wish to alienate yourselves from them, please contact us and we will handle the situation with complete confidentiality.
       1. Please see the code for enforcement below.

</details>

<p align="right">(<a href="#top">back to top</a>)</p>

## Enforcement

When enforcing the Code of Conduct, especially in cases involving non-native English speakers, we encourage patience, empathy, understanding, and willingness to educate and inform. In context where the breach is deemed *intentional*, we will pursue enforcement of the Code of Conduct as described.

> *"People from different backgrounds are likely to have different interpretations of what words or behaviors are considered acceptable. It's somewhat ironic that in an attempt to draw a more diverse developer base, code of conduct moderation, if improperly conducted, could potentially alienate users from other countries who don't have the same social conditioning as those in North America [or other Western cultures]."*[^2]

> 1. The Code of Conduct can be enforced through a range of passive and active means. We recommend minimal use for first offenders but allow rapid escalation if required (depending on the severity of the breach).

Reviewer: What is "minimal use"?

Author: As according to the reasonable person principle; we have not yet discussed hierarchy of rule enforcement so I cannot determine this without input from everyone.

Reviewer 2: I'm ok with it, gives room to grow.

<div id="top"></div>

<br />

<div align="center">
    <a href="https://github.com/nim-works">
        <img src="https://raw.githubusercontent.com/nim-lang/assets/master/Art/logo-crown.png" height="80px"/>
    </a>
    <h1 align="center">Code of Conduct</h3>
    <p align="center">
    Last Revision - 06/12/2021
    </p>
</div>

<br />

### Purpose

> We want to make sustainable technology with a diverse and cohesive community.
> In the same way that we follow best-practices of software engineering based
> on research and community experiences, we follow the latest evidence in the
> inclusion of a Code of Conduct (CoC) and its design. We welcome feedback.

Reviewer 2: ```suggestion
We want to make sustainable technology, this requires a community that is
architected along the same principle. The Code of Conduct (CoC) is a key tool
in ensuring a community that is diverse in experience, cohesive in practice,
and growing in membership. We welcome feedback.
```

We empathize with serious concerns that CoC may represent a loss of freedom of
speech, inhibit personal expression, or bring unnecessary discussion to what
should be the development of technology.

To these concerns, we say that evidence-based research has shown that Codes
of Conduct are instrumental to achieving and sustaining inclusive communities
which promote diverse sets of skills and perspectives. Meritocracy is not
sustainable outside of interactions between absolute equals, including cultural
norms and understandings.

</details>

<p align="right">(<a href="#top">back to top</a>)</p>

## Scope

This Code of Conduct applies to spaces both online and offline.

<details>

Online spaces managed by the Nim-Works organization will be held to the
following standards. Spaces that represent our users, or where users are
otherwise representing Nim-Works, may still be held accountable to these
standards depending upon context and official status of the representative.

Offline spaces where members interact, including but not limited to
conferences, will be similarly held to the standards below.

</details>

<p align="right">(<a href="#top">back to top</a>)</p>

## Code of Conduct

> Simplicity is best. However, we recognize that "simple" is a subjective and
> non-inclusive term itself. Therefore, we have outlined the main values of our
> Code of Conduct; however, you can expand the sections below particular values
> to see more bullshit to puzzle over.

Reviewer 2: ```suggestion
Outlined below are the main values of the Code of Conduct, each of which can be
expanded to show contributing rules and themes:
```

Less is more?

   1. Discrimination against an individual(s) or group(s) damages the community and attacks the goals of our contributions; we specifically will not discriminate based on:
      1. Virtue of race or ethnicity; country of origin, or country of residence; gender(s), biological sex or sexual orientation; age or experience; economic standing or education; disability - physical or otherwise. Not listed are not excluded.
   2. Harassment - online or otherwise are not tolerated. This includes any offensive, belittling or threatening behaviour toward an individual(s) or group(s).
      1. Cultural diversity and language barriers may play a critical role in differentiating whether a perpetrator intentionally meant to be aggressive in their behaviour. Patience is crucial in investigating these cases.

</details>

2. <details><summary>Contribution does not entitle me or anyone else to act in any way that is not conducive to making this a productive and enjoyable environment</summary><br />

    1. Status within the organisation or technical skill does not allow someone to be exempt from the rules.

>        1. The community does not owe any individual, irregardless of the size of their contributions or financial investments.

Reviewer 2: ```suggestion
       1. The community does not owe any individual, regardless of the size of their contributions or financial investments.
```

I just don't like `irregardless`.

    2. Members with greater responsibility (eg: moderators) are held to much higher expectations than others. We believe in leading by example.

>        1. The status of a community moderator is a privilege, but also a responsibility.

Reviewer 2: ```suggestion
       1. Community moderator status is both a privilege and responsibility.
```

       2. Moderators may step down if they do not feel they are able to uphold these responsibilities.
    3. We should be patient with new or less experienced members; our contributions do not entitle us to belittle others (which would be indicative of harassment/discrimination)

</details>

> 3. <details><summary>We welcome new users; whether they be contributors, testers, documentation editors, or someone interested in our work. We will help those who need a hand getting started (within reason) with the same patience we would expect someone to give us.</summary><br />

Reviewer 2: ```suggestion
3. <details><summary>We welcome new users; whether contributor, tester,
   documentation editor, or someone interested in our work. We will help those
   who need a hand getting started (within reason) with the same patience we
   would expect someone to give us.</summary><br />
```

wrapped, trimmed back a bit.

4. Moderators must assemble evidence for punishments enacted should they be requested to provide it.
   1. The moderator is not required to include their interpretation or deliberation of the material. This however may be raised by the aggrieved party (or a representative) in which case the decision will be reviewed within 7 days by another moderator who is independent of the first.

<!-- If a member of the community or public is ever concerned with any actions of our moderation or code of conduct, they may let any of our moderators know. Should they feel this is inadequate or inappropriate for whatever reason, they may communicate with @shayanhabibi. Communications will be confidential unless told otherwise. -->

<p align="right">(<a href="#top">back to top</a>)</p>

<!-- ## Revision

Revision of this document would require a democratic decision involving the majority of active Nim-Works organisational members within 7 days of the revision proposed. This does not apply when the revision does not meaningfully change the content within (ie it is related to formatting) -->

## References

> 1.	Singh V, Bongiovanni B, Brandon W. Codes of conduct in Open Source Software—for warm and fuzzy feelings or equality in community? Software Quality Journal. 2021.
> 
> 2.	Li R, Pandurangan P, Frluckaj H, Dabbish L. Code of Conduct Conversations in Open Source Software Projects on Github. Proceedings of the ACM on Human-Computer Interaction. 2021;5(CSCW1):1-31.
> 
> 3.	Dunbar-Hester C. Hacking Technology, Hacking Communities: Codes of Conduct and Community Standards in Open Source. MIT Case Studies in Social and Ethical Responsibilities of Computing. 2021.
> 
> 4.	Tourani P, Adams B, Serebrenik A, editors. Code of conduct in open source projects 2017: IEEE.

Reviewer 2: ```suggestion
[^1]:	Singh V, Bongiovanni B, Brandon W. Codes of conduct in Open Source
      Software—for warm and fuzzy feelings or equality in community? Software
      Quality Journal. 2021.

[^2]:	Li R, Pandurangan P, Frluckaj H, Dabbish L. Code of Conduct Conversations
      in Open Source Software Projects on Github. Proceedings of the ACM on
      Human-Computer Interaction. 2021;5(CSCW1):1-31.

[^3]:	Dunbar-Hester C. Hacking Technology, Hacking Communities: Codes of
      Conduct and Community Standards in Open Source. MIT Case Studies in
      Social and Ethical Responsibilities of Computing. 2021.

[^4]:	Tourani P, Adams B, Serebrenik A, editors. Code of conduct in open
      source projects 2017: IEEE.
```

Github footnotes; then these can be referenced in the text above via `[^1]` etc.

So in the beginning when talking about evidence based research popping in the refs will create little superscripts links to the bottom here.
